### PR TITLE
make prune to remove dangling images only

### LINKED ISSUE #1922: Make `image prune` remove dangling images by default
Currently `nerdctl image prune` requires `--all` to be specified. Since dangling image is supported, we should remove this restriction and remove dangling images by default and remove all unused if `--all` is specified.

`nerdctl image prune`: https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#whale-nerdctl-image-prune

dangling support: https://github.com/containerd/nerdctl/pull/1919

`docker image prune`: https://docs.docker.com/engine/reference/commandline/image_prune/

### PR DESCRIPTION
**Pull request by suyanhanx**
close #1922 

fix
- #1922

Signed-off-by: Han Xu <suyanhanx@gmail.com>

### CODE REVIEW

File: cmd/nerdctl/image_prune.go
Comment (manugupt1) [inline cmd/nerdctl/image_prune.go:70]: nit, `msg =`

File: cmd/nerdctl/image_prune.go
Comment (manugupt1) [inline cmd/nerdctl/image_prune.go:72]: nit, `msg =`

File: cmd/nerdctl/image_prune.go
Comment (suyanhanx) [reply cmd/nerdctl/image_prune.go]: Thanks. Done.

**Comment by suyanhanx** [comment]
@djdongjin PTAL, thanks.

File: cmd/nerdctl/image_prune.go
Comment (djdongjin) [inline cmd/nerdctl/image_prune.go:131]: nit: after either this PR or #1931 is merged, maybe good to rebase the other and use the existing `filterDangling` (https://github.com/containerd/nerdctl/pull/1931#discussion_r1088125051).

**Review by djdongjin — APPROVED** [review]
LGTM, thanks

File: cmd/nerdctl/image_prune.go
Comment (suyanhanx) [reply cmd/nerdctl/image_prune.go:131]: Yep. I'm waiting to do this.

Comment (suyanhanx) [reply cmd/nerdctl/image_prune.go]: Rebased. @djdongjin

File: pkg/cmd/image/list.go
Comment (djdongjin) [inline pkg/cmd/image/list.go:124]: can we keep the func here? Since there are other image filter funcs here (filterByReference, filterByLabel).

Comment (suyanhanx) [reply pkg/cmd/image/list.go]: Sure. Done.

**Review by djdongjin — APPROVED** [review]
LGTM, thanks

File: cmd/nerdctl/image_prune_test.go
Comment (manugupt1) [inline cmd/nerdctl/image_prune_test.go:56]: Can you replace Run with AssertOk. If possible in the previous test as well.

**Review by manugupt1 — COMMENTED** [review]
LGTM after replacing Run with AssertOk

File: cmd/nerdctl/image_prune_test.go
Comment (suyanhanx) [reply cmd/nerdctl/image_prune_test.go]: I see that the vast majority of commands executed with defer are using `Run`, and that is also what I have been doing. For other commands, I have been using `AssertOK`. Perhaps we could uniformize this approach throughout the entire project later.

Updated.

Comment (manugupt1) [reply cmd/nerdctl/image_prune_test.go]: It should be updated; we just change as we go; sometimes Run fails; and our  tests get flaky

**Review by manugupt1 — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**
